lib/modules/platform/github/index.ts

```diff
@@ -166,14 +170,31 @@ export async function initPlatform({
     }
   }
   logger.debug({ platformConfig, renovateUsername }, 'Platform config');
-  const platformResult: PlatformResult = {
+  const githubDS: HostRule[] = [];
+  const gitHubRule: HostRule[] = [
+    {
+      matchHost: 'ghcr.io',
+      username: renovateUsername,
+      password: token,
+    },
+  ];
+  if (!platformConfig.isGhe) {
```

Comment:
Shouldn't this be for all rules? e.g. why would we push a hostrule for a private GHE instance for ghcr.io?

Reply:
didnt know there's a GHE ghcr.io instance ill add it

Reply:
?? there is no `ghcr.io` for GHES. thats why we can only add the rules for `github.com` endpoint